harp/reader.py

```diff
@@ -75,6 +80,245 @@ def __dir__(self) -> Iterable[str]:
     def __getattr__(self, __name: str) -> RegisterReader:
         return self.registers[__name]
 
+    @staticmethod
+    def from_file(
+        filepath: PathLike,
+        base_path: Optional[PathLike] = None,
+        include_common_registers: bool = True,
+        epoch: Optional[datetime] = None,
+        keep_type: bool = False,
+    ) -> "DeviceReader":
+        """Creates a device reader object from the specified schema yml file.
+
+        Parameters
+        ----------
+        filepath
+            A path to the device yml schema describing the device.
+        base_path
+            The path to attempt to resolve the location of data files.
+        include_common_registers
+            Specifies whether to include the set of Harp common registers in the
+            parsed device schema object. If a parsed device schema object is provided,
+            this parameter is ignored.
+        epoch
+            The default reference datetime at which time zero begins. If specified,
+            the data frames returned by each register reader will have a datetime index.
+        keep_type
+            Specifies whether to include a column with the message type by default.
+
+        Returns
+        -------
+            A device reader object which can be used to read binary data for each
+            register or to access metadata about each register. Individual registers
+            can be accessed using dot notation using the name of the register as the
+            key.
+        """
+
+        device = read_schema(filepath, include_common_registers)
+        if base_path is None:
+            path = Path(filepath).absolute().resolve()
+            base_path = path.parent / device.device
+        else:
+            base_path = Path(base_path).absolute().resolve() / device.device
+
+        reg_readers = {
+            name: _create_register_parser(
+                device, name, _ReaderParams(base_path, epoch, keep_type)
+            )
+            for name in device.registers.keys()
+        }
+        return DeviceReader(device, reg_readers)
+
+    @staticmethod
+    def from_url(
+        url: str,
+        base_path: Optional[PathLike] = None,
```

Comment:
`base_path` is indeed getting clunky, I'm slightly uncomfortable of having different behaviors depending on which function you use here, so maybe just getting rid of it would be best.

Reply:
Not sure I follow. The use of `base_path` is consistent across methods. If provided it will try use it as a root directory where all files are expected to be found. If not provided it will assume either the current working directory (e.g. `from_url`) or the root of the provided `yml` (e.g. `from_file`). Are these two defaults what are raising flags for you? Happy to go with a different one if you want. We can also just throw an error at the level of the `DeviceReader` class if no path was provided and the user try to use the no-input overload of the register load function.